docs/design/windows-images.md

```diff
@@ -0,0 +1,164 @@
+# Building Azure marketplace images for Windows AKS support
+
+## Motivation
+
+Initially, AKS-Engine's codebase on Linux installed all of the needed stuff from package repos or container registries. This provided the best flexibility as any packages could be easily updated and tested with a simple generate+deploy. However, it had a few major downsides. Network or repo downtime would cause failed deployments that were not recoverable. It also took additional time to do the downloads, making the overall deployments slower. To mitigate this while preserving deployment flexibility, AKS-engine switched to using custom VM images [built with Packer](../../packer). 
+
+Windows is in a halfway state today were the Windows Server team publishes `Windows Server version ___ with Containers` that has Docker preinstalled and some container images `nanoserver` and `windowsservercore` pre-pulled. This helps save some deployment time, but means that making changes to it depends on another team, and not having automation makes it much more difficult to pre-test changes to what's installed in these images. We need the flexibility to validate new OS versions, patches, and other dependencies. Moving to an image-based solution would align with the Linux workflow and allow us to control and pre-test all changes. It also leaves the option to make other customizations specific to AKS that will help with reliability or deployment time.
+
+The other challenge with the Windows deployment workflow today is that all of the Kubernetes node binaries, CNI plugin, and default configuration files are in a single ZIP published to the acsmirror CDN. It's built + packaged in [build-windows-k8s.sh](../../scripts/build-windows-k8s.sh). This was written when the upstream kubernetes/kubernetes code was in an alpha state, downstream patches were required, and public builds were not yet available. This is no longer needed and makes it harder to revise and test those components individually.
+
+All of this needs to be cleaned up so that we can get avoid deployment time penalties due to downloads and keep all changes in dependencies in source code so that they can be thoroughly tested and reproduced. 

```

Comment:
For Linux, our base image is not available in all sovereign clouds (I assume it will be the same for Windows). When modifying the code to work with the VHD, we should always maintain a working installation path for clusters not using the VHD (by choice or by necessity). Also, keeping an install path for all components allows for faster development speed because we don't need to wait for a new image to be published (which can take a few days) to be able to test changes to the download code flows (eg. new component versions).